### PR TITLE
refactor: split words by language

### PR DESCRIPTION
**Pull request by davideprevosto**
This Pull Request splits words by language. It also adds a first version of the Italian stop-words configuration.

### CODE REVIEW

**Comment by JonPurvis** [comment]
Hey @davideprevosto 

Would you be able to take a look at the failing static analysis checks? From what I can see:

1) `composer lint` needs running 
2) You've moved the config files into `Config/words/` rather than just `Config/` but when including them, not looking within /words. I think they should just live in `Config/` though like before.

Thanks!

**Comment by faissaloux** [comment]
@JonPurvis I think the words files should live in a separated directory in Config to separate them from `tolerated` file like @davideprevosto did this directory may have multiple files later on for each language, I'm just thinking to rename the dir `words` to `profanities` 🤔

**Comment by JonPurvis** [comment]
Ah yes good point @faissaloux, somehow I forgot we had tolerated.php now 😆 (have updated my comment)

Agreed on a directory name change, but how about `lang`?

**Comment by faissaloux** [comment]
> Agreed on a directory name change, but how about `lang`?

Emmmmm `lang` doesn't represent what's inside. I think `profanities` is better!

If someone gets there will know exactly what `en.php` is for exactly from its directory name (`profanities`)

**Comment by JonPurvis** [comment]
`profanities` it is 😄

File: src/Expectations/Profanity.php [hunk 1]
```diff
@@ -10,7 +10,11 @@
 expect()->extend('toHaveNoProfanity', fn (array $excluding = [], array $including = []): ArchExpectation => Targeted::make(
     $this,
     function (ObjectDescription $object) use (&$foundWords, $excluding, $including): bool {
-        $words = include __DIR__.'/../Config/words.php';
+        $words = array_merge(
+            include __DIR__.'/../Config/en.php',
+            include __DIR__.'/../Config/it.php',
+        );
```
Comment (faissaloux) [inline src/Expectations/Profanity.php:16]:
```suggestion
        $words = [];
        $profanitiesDir = __DIR__.'/../Config/profanities';

        if (($profanitiesFiles = scandir($profanitiesDir)) === false) {
            return true;
        }

        $profanitiesFiles = array_diff($profanitiesFiles, ['.', '..']);

        foreach($profanitiesFiles as $profanitiesFile) {
            $words = array_merge(
                $words,
                include "$profanitiesDir/$profanitiesFile"
            );
        }
```

**Review by faissaloux — CHANGES_REQUESTED** [review]
Can we please make it more flexible to accept more languages by just adding more languages files on the `profanities` directory without adding languages files to the code one by one.

File: src/Config/words/it.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+return [
```
Comment (faissaloux) [inline src/Config/words/it.php:5]:
```suggestion
// SHOULD ALL WORDS BE lowercase!

return [
```

**Review by faissaloux — CHANGES_REQUESTED** [review]
Can you please add this note `SHOULD ALL WORDS BE lowercase!` to all words files on line 5

**Comment by JonPurvis** [comment]
Hey @davideprevosto 👋 

I'm keen to get this work into the package. Would you have any time soon to action the required changes? If not, I'd be happy to commit to this branch to finish the work off and get it merged 😄

**Comment by davideprevosto** [comment]
Hi @JonPurvis, I tried to follow your instructions and I pushed a few changes. I run composer lint and I suppose Laravel Pint did its job.

I also run composer test but I run into the following issue, I suppose I need to further investigate what I am doing wrong.

```
composer test
> rector --dry-run
sh: /Users/xxx/Library/Application: No such file or directory
Script rector --dry-run handling the test:refacto event returned with error code 127
Script @test:refacto was called via test
```

**Comment by JonPurvis** [comment]
Hey @davideprevosto 

Not actually sure what could be causing that error. Do you get the same thing when you run `composer test:refacto`?

**Comment by davideprevosto** [comment]
> composer test:refacto

Yes @JonPurvis, I am getting the same error

```
composer test:refacto
> rector --dry-run
sh: /Users/xxxx/Library/Application: No such file or directory
Script rector --dry-run handling the test:refacto event returned with error code 127
```

**Comment by JonPurvis** [comment]
Okay, can you try running `composer test:lint`? I want to see if all of them have the same issue or it's just refacto

**Comment by davideprevosto** [comment]
> Okay, can you try running `composer test:lint`? I want to see if all of them have the same issue or it's just refacto

image

**Comment by JonPurvis** [comment]
Hmm, that's odd. It works for me even with your changes. Although it does fix one thing:

```php
    FileLineFinder::where(function (string $line) use (&$foundWords): bool {
        return str_contains(strtolower($line), strtolower(array_values($foundWords ?? [])[0]));
    })
```

to

```php
    FileLineFinder::where(function (string $line) use (&$foundWords): bool {
        return str_contains(strtolower($line), strtolower((string) array_values($foundWords ?? [])[0]));
    })
```

If you make that change, and then amend the merge conflict, this branch should be good to merge. I'm not sure why refacto doesn't run properly on your machine 😢

**Comment by JonPurvis** [comment]
Thanks @davideprevosto 

Are you also able to sort the merge conflict please?

**Comment by davideprevosto** [comment]
@JonPurvis can you help me with the merge conflict please? 

image

I tried to merge the branch into the master branch (on my fork) without issues.

**Comment by JonPurvis** [comment]
Thanks for the contribution @davideprevosto 🥳 I'll get a new release tagged later today!

**Pull request merged**